### PR TITLE
Improve extraction of GitHub issue data

### PR DESCRIPTION
**Pull request by bockthom**
In this pull request, I add lots of improvements and take of more consistency of the issue data.

This pull request is related to se-passau/GitHubWrapper#25.
This pull request should only be merged as far as se-passau/GitHubWrapper#25 is approved, at least (or merged in the end).

Here are the main enhancements:

- Add review data, review comments, review requests, review dismiss events [94bab3a, 4eb6e20, 
b4283f9, 3862355, 16debcf]
- Unify GitHub users, that is, bring usernames and name/e-mail of commits together [9370972, 
973e2b9, 1faf340, 47e3ec8, 6adb906]
- Fix and enhance information for "assigned", "unassigned", "mentioned", "subscribed" events [
622b7f8, 8d66e45, b135036]
- Distinguish related commits in "add_link", "comitt_added", and "referenced_by" [39ff302, 2254fb9, 2b5e72b, 5602a7f ]
- Add a reversed event for referencing issues, that is, "referencedy_by" as opposite of "add_link" [c34842a, 2b5e72b]
- Remove duplicated events [b9a7182]
- Automatically add a "commented" event for each "created" event [18a65da, 066d875]

As soon as my enhancements are approved, I will upload an updated version of the diagram which shows all the events and their possible fields.

Please let me know if you have any suggestions on how to further improve the issue data.

### CODE REVIEW

File: issue_processing/issue_processing.py
Comment (clhunsen) [inline issue_processing/issue_processing.py]: Can we refactor this into a separate lookup function? Then, we could pass `event["user"]` and `event["ref_target"]` and return the altered or unaltered user "object".

File: issue_processing/issue_processing.py
Comment (clhunsen) [inline issue_processing/issue_processing.py]: Can we put this String constant into a file-local constant?

**Review by clhunsen — APPROVED** [review]
Wow, what an extension of the GitHub-related functionality. :open_mouth: I only have suggestions for refactorings, otherwise the code looks good to me.

File: issue_processing/issue_processing.py
Comment (bockthom) [reply issue_processing/issue_processing.py]: Yes, sounds like a good idea!

File: issue_processing/issue_processing.py
Comment (bockthom) [reply issue_processing/issue_processing.py]: Done.

**Comment by bockthom** [comment]
If have added some further improvements on this PR now.

Actually, I extracted the data from this PR for evaluating some special cases, resulting in some additional corner cases :wink: 

The initial comment of this PR leads to another problem which we have to think about how to deal with in the GitHubWrapper: In the initial comment, I have referenced issue 25 from the GitHubWrapper. However, by the GitHubWrapper, this is recognized as a link from codeface-extraction 26 (this PR) to codeface-extraction 25 (which actually is not referenced). Sorry for not referencing those things in this comment - that would lead to lots of references in the data of this issue, which I wanted to avoid for debugging reasons. To mitigate this issue, I suggest the following solutions to be solved in the GitHubWrapper: 
- Check if there is a space before `#` or this sign is at the beginning of a line. 
- If not, check if the preceding prefix ends with the name of the repository (as far as this is available in the GitHubWrapper while issue extraction...)

**Comment by bockthom** [comment]
I have run the issue extraction on the complete OpenSSL data and the result look good (at least, from what I have spot-checked). So, from my point of view, this PR is ready to merge if there are no further change requests. 
[Please postpone merging this PR until se-passau/GItHubWrapper#25 is merged.]

**Comment by clhunsen** [comment]
Sound really good, @bockthom! :+1: The code looks good to me; @ecklbarb, @fehnkera, are there any remarks from your side?

**Comment by bockthom** [comment]
I have updated the corresponding diagram now:
[issue_data_processing.pdf](https://github.com/se-passau/codeface-extraction/files/3450757/issue_data_processing.pdf)
Please have a look at it.

[Disclaimer: I only have mentioned the event types which occur in the recently extracted OpenSSL data, which is quite a lot (and even more than I have expected). There may be also other event types (which only rarely occur, though) in other projects.]

**Pull request merged**
